clifford/_layout.py

```diff
@@ -596,6 +844,38 @@ def left_complement_func(self):
     def right_complement_func(self):
         return self._gen_complement_func(omt=self.omt.T)
 
+    @_cached_property
+    def overload_mul_func(self):
+        return get_overload_mul(self)
+
+    @_cached_property
+    def overload_xor_func(self):
+        return get_overload_xor(self)
+
+    @_cached_property
+    def overload_or_func(self):
+        return get_overload_or(self)
+
+    @_cached_property
+    def overload_add_func(self):
+        return get_overload_add(self)
+
+    @_cached_property
+    def overload_sub_func(self):
+        return get_overload_sub(self)
+
+    @_cached_property
+    def overload_reverse_func(self):
+        return get_overload_reverse(self)
+
+    @_cached_property
+    def project_to_grade_func(self):
+        return get_project_to_grade_func(self)
+
+    @_cached_property
+    def overload_call_func(self):
+        return get_overload_call(self)
```

Comment:
I don't think these really belong here. You've put them here because it makes them easy to cache, but I think it would be better to cache them manually in a `weakref.WeakKeyDictionary`.

Reply:
So perhaps something like
```
def weak_cache(f):
    _cache = weakref.WeakKeyDictionary()
    @functools.wraps(f)
    def wrapped(*args, **kwargs):
        a, *args = args
        try:
            return _cache[a]
        except KeyError:
            ret =_cache[a] = f(a, *args, **kwargs)
            return ret
    wrapped._cache = _cache
    return wrapped


@weak_cache
def _get_jit_impls(layout):
    return {
        'as_ga': get_as_ga_value_vector_func(layout),
        'ga_add': get_overload_add_func(layout),
        'ga_sub': get_overload_sub_func(layout),
        'ga_mul': get_overload_mul_func(layout),
        'ga_xor': get_overload_xor_func(layout),
        'ga_or': get_overload_or_func(layout),
        'ga_rev': get_overload_reverse_func(layout),
        'ga_call': get_overload_call_func(layout),
    }
```